### DISCUSSION TITLE
Saving a receipe takes extremly long

### DISCUSSION THREAD

**Discussion by LimesRunner** [TandoorRecipes/recipes#2368] [Q&A]
I really like this application. I installed it on my Synology NAS following [this](https://github.com/vabene1111/recipes/files/6901601/Tandoor.on.a.Synology.Disk.Station.pdf) documentation on the Tandoor Receipes Website.
But I have one problem: When I create a new receipe or edit an existing one and click on "Save" or "Save & Watch" it lasts 35 - 45 seconds until the receipe is stored. The problem is also, that there is no feedback during that time, that the application is still working. I could also continue editing, which may lead to data loss.

Is that the usual behavior or is there something wrong with my installation? I run also other docker containers based on a database, e.g. paperless-ngx, and there I do not have such issues.

**Comment by vabene1111** [comment]
This is not normal and must be related to an issue caused by the setup since I have never heard of this before.

Do you have a really large collection of recipes (>1000)? Normally even than it should only take a second or a few max.

**Reply by LimesRunner** [reply]
No, I currently have less than 10 receipes in my data base. 
Do you have any idea how I can solve this? Is it possible, that the proposed postgres version 11-alpine is very slow?

When I want to open a receipe or surch for receipes, everything is as fast as it should be.

**Reply by vabene1111** [reply]
very interesting, i remember that there was a user a very long time ago who had similar problems on synology, you might want to look a bit trough old issues. 

its the default but also check that all tree features in .env are disabled.

**Reply by LimesRunner** [reply]
I followed the instruction provided by DiversityBug. Here, the env variables are set manually. I assume, as the tree feature variables are not set here, it should be disabled.

Maybe I give it a try to set up new containers, using docker compose and the .env-file.

**Reply by LimesRunner** [reply]
By the way, I had now time do set up a second instance from Tandoor, this time using a docker-compose and using the .env- file. 
The saving time now has reduced to around 15 - 18 seconds. It seems not to depend on puctures or ammount of steps.
Still not perfect, but much better compared to the original setup.
This is acceptable for me but I still wonder, why it lasts such a long time.

**Reply by vabene1111** [reply]
Very strange. Saving is a very complex operation as many database tables need to be updates but it should never take this long.

Is your CPU load increasing in any meaningful way?

In the past I have mostly focused on view performance optimization but maybe I need to look at editing at some point as well.

**Reply by LimesRunner** [reply]
The docker container is running on a Synology DS220+ with 10 GB RAM and an Intel Celeron J4025 2-Core 2,0 GHz.  During the saving time, there is a small peak in the CPU up to 35%. I think not, that this should be a problem. 

I have two other docker containers running, that are using a database: Paperless-ngx, which is also using a postgres-container (~2 s saving time) and Bookstack, which is using MariaDB (3 - 4 s saving time). I also tried out Mealie yesterday, which is using SQLite and is very very fast, but I like Tandoor much better for my needs. Maybe I will ask in the Synology Community, if someone has similar expirience.

Maybe it would be a benfit, if there would be at least the "loading screen" visible during the saving time (the screen, where the incredients clip-arts are turning in a circle). So that the user knows, that something is happening.

**Reply by vabene1111** [reply]
Yes I need to simulate slow speeds to test that.

I remember that there was a very old issue similar to this, but not sure if it ever got solved, would really be interested in finding out why this is so slow.

**Comment by ghost** [comment]
Hello to everyone,

I got the same problem with latest develop branch on Synology 920+ with 3 recipes in database. I have also loaded the default data for tandoor.

Environment file does include turning off debug and database settings as were set in the default env file.

Just give me a hint, if you need any further information or how I can help you otherwise, so we can get rid of this problem.

Btw. it takes about 30 seconds here for saving.

Greetings
Christian

**Comment by lagset** [comment]
Hey there, just wanted to note, that I have the same issue. Read and viewing in UI is quite snappy and quick (especially with tandoor 2 alpha, nice!). But saving is taking often more than 30 seconds.

Running on Synology DS420+ with 10GB RAM. No relevant CPU spikes observed, so far. Using docker-compose setup.

If it helps, I can of course contribute logs.
